src/memusage.h

```diff
@@ -90,6 +91,18 @@ static inline size_t DynamicUsage(const std::vector<T, Allocator>& v)
     return MallocUsage(v.capacity() * sizeof(T));
 }
 
+static inline size_t DynamicUsage(const std::string& s)
```

Comment:
Could you please add a test in this commit that demonstrates the intended behavior? Or would that tie the implementation down too much?

Reply:
i guess we could do some basic generalizable test like "the memory usage increases or stays equal when the string becomes longer"
but i'm not sure how useful that is, and there are currently no memusage tests to add it to

Reply:
i've tested it manually (gcc, x86_64) with the following patch:
<details>
  <summary>patch</summary>

```patch
diff --git a/src/bitcoin-cli.cpp b/src/bitcoin-cli.cpp
index 8a4c1746161c30c04dee1f41a811740d0038df0b..f8c688dd3617123548af13611519adba59eef775 100644
--- a/src/bitcoin-cli.cpp
+++ b/src/bitcoin-cli.cpp
@@ -23,6 +23,7 @@
 #include <util/strencodings.h>
 #include <util/time.h>
 #include <util/translation.h>
+#include <memusage.h>

 #include <algorithm>
 #include <chrono>
@@ -1281,6 +1282,13 @@ static int CommandLineRPC(int argc, char *argv[])

 MAIN_FUNCTION
 {
+    std::string s;
+    std::cout << "length (b)" << "," << "dynsize (b)" << std::endl;
+    for (int x=0; x<=200; ++x) {
+        std::cout << x << "," << memusage::DynamicUsage(s) << std::endl;
+        s += 'a';
+    }
+    exit(0);
 #ifdef WIN32
     common::WinCmdLineArgs winArgs;
     std::tie(argc, argv) = winArgs.get();
```

</details>

as expected, for the first 15 bytes it stays flat at 0, then it allocates and the first allocation is 48 bytes, and it increases stepwise as the size of the string grows:
![image](https://github.com/user-attachments/assets/7ba2adec-b0be-4e31-bafe-a8b9386af222)